bigchaindb/core.py

```diff
@@ -219,17 +221,26 @@ def get_owned_ids(self, owner):
             list: list of `txids` currently owned by `owner`
         """
 
-        response = r.table('bigchain')\
-                    .concat_map(lambda doc: doc['block']['transactions'])\
-                    .filter({'transaction': {'new_owner': owner}})\
-                    .pluck('id')['id']\
-                    .run(self.conn)
+        response = r.table('bigchain') \
+            .concat_map(lambda doc: doc['block']['transactions']) \
+            .filter(lambda tx: tx['transaction']['conditions']
+                    .contains(lambda c: c['new_owners']
+                              .contains(owner))) \
+            .run(self.conn)
         owned = []
 
-        # remove all inputs already spent
-        for tx_input in list(response):
-            if not self.get_spent(tx_input):
-                owned.append(tx_input)
+        for tx in response:
```

Comment:
The following part is a bit difficult to understand, a couple of ideas to make it simpler:
 - add some inline comments, using "natural language" to explain what's happening
 - restructure the code to lower the complexity (since we have an `if`, inside an `if`, inside a `for`, inside a `for`), maybe some of the parts can be split up into different funcs?